src/main/java/com/minecolonies/colony/buildings/BuildingFarmer.java

```diff
@@ -4,89 +4,411 @@
+     */
     public BuildingFarmer(Colony c, BlockPos l)
     {
         super(c, l);
     }
 
+    /**
+     * Returns list of fields of the farmer.
+     * @return a list of field objects.
+     */
+    public List<Field> getFarmerFields()
+    {
+        return Collections.unmodifiableList(farmerFields);
+    }
+
+    /**
+     * Checks if the farmer has any fields.
+     * @return true if he has none.
+     */
+    public boolean hasNoFields()
+    {
+        return farmerFields.isEmpty();
+    }
+
+    /**
+     * Assigns a field list to the field list.
+     * @param field the field to add.
+     */
+    public void addFarmerFields(Field field)
+    {
+        farmerFields.add(field);
+    }
+
+    /**
+     * Getter of the current field.
+     * @return a field object.
+     */
+    public Field getCurrentField()
+    {
+        return currentField;
+    }
+
+    /**
+     * Sets the field the farmer is currently working on.
+     * @param currentField the field to work on.
+     */
+    public void setCurrentField(final Field currentField)
+    {
+        this.currentField = currentField;
+    }
+
+    /**
+     * Retrieves a random field to work on for the farmer.
+     * @return a field to work on.
+     */
+    public Field getFieldToWorkOn()
+    {
+        Collections.shuffle(farmerFields);
+        for(final Field field: farmerFields)
+        {
+            if(field.needsWork())
+            {
+                currentField = field;
+                return field;
+            }
+        }
+        return null;
+    }
+
     @Override
-    public String getSchematicName(){ return FARMER; }
+    public String getSchematicName()
+    {
+        return FARMER;
+    }
 
     @Override
-    public int getMaxBuildingLevel(){ return 3; }
+    public int getMaxBuildingLevel()
+    {
+        return MAX_BUILDING_LEVEL;
+    }
 
     @Override
-    public String getJobName(){ return FARMER; }
+    public String getJobName()
+    {
+        return FARMER;
+    }
+
+    /**
+     * Synchronize field list with colony.
+     * @param world the world the building is in.
+     */
+    public void syncWithColony(World world)
+    {
+        if(!farmerFields.isEmpty())
+        {
+            final ArrayList<Field> tempFields = new ArrayList<>(farmerFields);
+
+            for(final Field field: tempFields)
+            {
+                final ScarecrowTileEntity scarecrow = (ScarecrowTileEntity) world.getTileEntity(field.getID());
+                if(scarecrow == null)
+                {
+                    farmerFields.remove(field);
+                    if (currentField != null && currentField.getID() == field.getID())
+                    {
+                        currentField = null;
+                    }
+                }
+                else
+                {
+                    scarecrow.setName(LanguageHandler.format("com.minecolonies.gui.scarecrow.user", getWorker().getName()));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "com.minecolonies.gui.scarecrow.user" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1192)